.github/workflows/ci-python.yml

```
@@ -24,7 +24,23 @@ jobs:
         with:
           packages: autohooks tests
           version: ${{ matrix.python-version }}
-
+  type-checking:
+    name: Type-checker
+    runs-on: 'ubuntu-latest'
+    strategy:
+        matrix:
+          python-version:
+            - 3.7
+            - 3.8
+            - 3.9
+            - "3.10"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: greenbone/actions/mypy-python@v1
```

Should be v2 at the end because new things will get merged from the main into the v2 branch now.
```suggestion
        uses: greenbone/actions/mypy-python@v2
```